### PR TITLE
Full screen support

### PR DESCRIPTION
**Pull request by rmichaelis**

### CODE REVIEW

**Comment by rmichaelis** [comment]
Tests in Travis failed due to a librabry version conflict.

Please review.

File: less/theme.less [hunk 1]
```diff
@@ -93,6 +93,16 @@
         background-color: @secondary-color;
       }
     }
+    
+    .ol-control button {
+        background-color: White;
```
Comment (pgiraud) [inline less/theme.less]: Though it works, I wouldn't use a capital case for colors.

**Comment by pgiraud** [comment]
@rmichaelis Can you please rebase your branch onto master?

**Comment by pgiraud** [comment]
@rmichaelis What command are you using for the rebase? I see weird things in the network tab.

**Comment by rmichaelis** [comment]
Yes, I saw it too. 
After a successfull rebase, git didn't accept that I push the branch. 
Then I tried to solve the problem with a pull as GIT suggested to me.

I try to fix it asap

**Comment by pgiraud** [comment]
When you do a rebase you usually change the git history because you change
commits. Well the commits should look the same but their id changes.
So most of the time (if not always) when you do a rebase you need to "push
--force" but that's expected.

On Tue, Jan 27, 2015 at 1:33 PM, rmichaelis <notifications@github.com>
wrote:

> Yes, I saw it too.
> After a successfull rebase, git didn't accept that I push the branch.
> Then I tried to solve the problem with a pull as GIT suggested to me.
>
> I try to fix it asap
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Geoportail-Luxembourg/geoportailv3/pull/177#issuecomment-71640828>
> .
>



-- 
Pierre GIRAUD
Géomaticien, Analyste

Camptocamp France SAS
Savoie Technolac, BP 352
73377 Le Bourget du Lac, Cedex

Tel : 00 33 4 79 44 44 93
Mail : pierre.giraud@camptocamp.com
http://www.camptocamp.com

**Comment by rmichaelis** [comment]
Thanks. 
It looks better now.

**Comment by pgiraud** [comment]
@rmichaelis Better.
Would you be kind to squash (or fixup) your commits into a single one to avoid the "fix merge conflicts" commits?
https://help.github.com/articles/about-git-rebase/#rebasing-commits-against-a-point-in-time

**Comment by rmichaelis** [comment]
@pgiraud Done.

File: geoportailv3/static/js/maincontroller.js [hunk 2]
```diff
@@ -82,6 +84,15 @@ app.MainController.prototype.setMap_ = function() {
 
   /** @type {ol.Map} */
   this['map'] = new ol.Map({
+    controls: [
+      new ol.control.Zoom({zoomInLabel: '', zoomOutLabel: ''}),
+      new ol.control.ZoomToExtent(
+          {extent:
+                [425152.9429259216, 6324465.99999133,
+                 914349.9239510496, 6507914.867875754]
+          }),
```
Comment (pgiraud) [inline geoportailv3/static/js/maincontroller.js]: These extent coordinates should go into a global value or constant if reused elsewhere.
We don't need it now though.

File: geoportailv3/static/less/layout.less [hunk 3]
```diff
@@ -90,3 +90,70 @@ body {
...
+.ol-control button {
+    font-size: 1.5em;
+    font-weight:normal;
+    text-align: center;
+    text-indent: 0;
+    border: 1px solid #8394a0;
+    font-family: "geoportail-icons" !important;
+    margin: -1px;
+}
+
```
Comment (pgiraud) [inline geoportailv3/static/less/layout.less]: What about taking advantage of less here?
The above could be simplified to something looking like the following:

```less
.ol-control {
  &, &:hover { // not tested
    background-color: transparent;
  }
  button {
    ...
  }
}
```

File: geoportailv3/static/less/layout.less [hunk 4]
```diff
@@ -90,3 +90,70 @@ body {
...
+.ol-zoom-extent {
+    left: 0.5em;
+    top: 88px;
+}
+
+.ol-zoom-extent button:after {
+    content: "\e01b";
+}
+
```
Comment (pgiraud) [inline geoportailv3/static/less/layout.less]:
```less
.ol-zoom-extent {
  ...
  button:after {
    ..
  }
}
```

**Comment by pgiraud** [comment]
I just tested it. Curiously in Chrome the map doesn't display in full width. Changing manually `width: auto` to `width: 100%` in the dev tools did the trick.

File: geoportailv3/static/less/layout.less [hunk 5]
```diff
@@ -90,3 +90,66 @@ body {
     bottom: 0;
   }
 }
+
+.ol-control button {
+    width:41px;
+    height:41px;
+}
+
```
Comment (pgiraud) [inline geoportailv3/static/less/layout.less]: Can't this be grouped with the code below ?
Also for consistency add a space after colon.

File: geoportailv3/static/less/layout.less [hunk 6]
```diff
@@ -90,3 +90,66 @@ body {
...
+    font-size: 0.8em;
+}
+
+.ol-overlaycontainer-stopevent {
+    left: 33px;
+    top: 80px;
+    position: absolute;
+}
+
+div.ol-zoom-extent > button {
```
Comment (pgiraud) [inline geoportailv3/static/less/layout.less]: Do you really need to be that explicit?

File: geoportailv3/static/less/theme.less [hunk 7]
```diff
@@ -93,6 +93,16 @@
         background-color: @secondary-color;
       }
     }
+    
+    .ol-control button {
+        background-color: white;
+        color: @primary-color;
+    }
+    .ol-control button:hover,.ol-control button:focus {
```
Comment (pgiraud) [inline geoportailv3/static/less/theme.less]: a line break here may help for readability

**Comment by pgiraud** [comment]
The `defaultExtent` constant looks good to me.

**Comment by pgiraud** [comment]
Please leave the 100% issue for now. Let's open a new issue for this.

**Comment by pgiraud** [comment]
Can you please rebase the branch onto master once again?

**Comment by pgiraud** [comment]
Also, the full screen button shouldn't be displayed on small devices. See how the print button style is handled.

File: geoportailv3/static/less/layout.less [hunk 8]
```diff
@@ -90,3 +90,43 @@ body {
     bottom: 0;
   }
 }
+
+.ol-control {
+    &, &:hover { // not tested
```
Comment (pgiraud) [inline geoportailv3/static/less/layout.less]: you should remove this comment  ;-)

Comment (rmichaelis) [inline geoportailv3/static/less/layout.less]: Oops :-)

**Comment by elemoine** [comment]
I've just added other comments. Otherwise looks good to me.

**Pull request merged**